org-code-javabuilder/lib/src/main/java/org/code/javabuilder/JavaProjectFile.java

```diff
@@ -43,10 +45,23 @@ public void setFileName(String fileName) {
 
   @Override
   public void setFileContents(String fileContents) {
-    this.fileContents = fileContents;
+    this.fileContents = this.importSystem(fileContents);
   }
 
   public void setClassName(String className) {
     this.className = className;
   }
+
+  private String importSystem(String fileContents) {
+    // Check if file contents start with system import only because system import should always
+    // be done by default by us, and therefore be the first thing in the file.
+    // We still want to import System in the rare case where SYSTEM_IMPORT is in
+    // a String somewhere in fileContents. In addition, a duplicate import will
+    // not cause any error to the student, it just makes the file longer.
```

Comment:
Why not just always prepend the import? i.e. even if it's already the first line?

Reply:
There isn't much risk to always prepending, the thought here was not to do it if it's not strictly necessary in case we end up somehow calling `setFileContents` a bunch of times. But I can revert it back to just always adding it at the beginning.

Reply:
thinking more on this, I'll just always add the import line, it's pretty low risk